backend/src/services/journal/journal.repo.js

```diff
@@ -115,5 +115,93 @@ class JournalRepo {
     });
     return entry;
   }
+
+  async getJournalTeamId(journalId) {
+    const entry = await this.getJournalById(journalId);
+    if (!entry) {
+      return null;
+    }
+    const user = await this.db.enrollments.findFirst({
+      where: {
+        user_id: entry.user_id,
+      },
+    });
```

Comment:
[Suggestion]
We could use the `loadCourse` preHandler and then pass the enrollment object here. This would reduce coupling.